examples/bitcoin_crate/src/main.rs

```diff
@@ -1,156 +1,93 @@
+                vout: 0,
+            },
+            script_sig: ScriptBuf::from_hex("47304402200552d42d255d9814bda2e0fbf28be6321a80615e633afb21d18abf4ad0bc721602202530e3f8e4bbf184eecfd95c615cb22ad06860fd75ce3aa423d0fea894a80332812103739ea9368ff2b1fdc4db2f160191e980190367501cc2b0c93a566fababd01064").unwrap(),
+            sequence: Sequence::from_hex("0xffffffff").unwrap(),
+            witness: Witness::default()
+        },
+        TxIn {
+            previous_output: OutPoint {
+                txid: Txid::from_str("e1926a751f36c25541263ca4b621f1e3376e15117c170e60ba82b113bda3cba6").unwrap(),
+                vout: 0,
+            },
+            script_sig: ScriptBuf::from_hex("47304402203743aeedace4ec68892c825ed674234b7c834268304aaaf76958d2da5670a9b702207b01a50a5c047023ac5a5b90367b1bd862ac6a4914b023a7c5463b3f8e3690710141047146f0e0fcb3139947cf0beb870fe251930ca10d4545793d31033e801b5219abf56c11a3cf3406ca590e4c14b0dab749d20862b3adc4709153c280c2a78be10c").unwrap(),
+            sequence: Sequence::from_hex("0xffffffff").unwrap(),
+            witness: Witness::default()
+        },
+        TxIn {
+            previous_output: OutPoint {
+                txid: Txid::from_str("0c20171f1558eb6c032a5916e8ce0b446fef6f7b102b2aabcfc83a8367b0f53b").unwrap(),
+                vout: 2,
+            },
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::from_hex("0xfffffffe").unwrap(),
+            witness: Witness::from_slice(&[
+                "30440220079287854aed2500913921a7b9698cefc17fc68fb02728d2e86c52173b6af6ba0220669efe80b73204efcda6c3a375da5fd059977e6c86bad977b15c562219cc5dd601",
+                "032d49f270684da5a422df37bf9818aa178ad89a975643c4928444aed78a7dd341"
+            ]),
+        },
+        TxIn {
+            previous_output: OutPoint {
+                txid: Txid::from_str("f2ccf54fa95a13b092d4f90c9cb215d11c858fd2964a45f3f997b3d4815e24e4").unwrap(),
+                vout: 1,
+            },
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::from_hex("0xfffffffd").unwrap(),
+            witness: Witness::from_slice(&[
+                "30450221008c10e822049f4e67388cd9dc825d1bf14c7f8b73801bc025e7534b33319d510f02205c214beb89a3b3e0837b6cc717dbda4f8707076050197b870cffec7594040c8801",
+                "02f2a0bdce72551e68dfbebf81d770924836cbb256a63e9987ea869eabac4859c1"
+            ]),
+        }
     ]
 }
 
-fn create_outputgroup(tx: Vec<Transaction>) -> Vec<OutputGroup> {
-    // Create OutputGroup from transaction data
-    tx.into_iter()
-        .enumerate()
-        .map(|(i, tx)| OutputGroup {
-            value: tx.output.iter().map(|op| op.value.to_sat()).sum(),
-            weight: tx.total_size() as u64,
-            input_count: tx.input.len(),
-            creation_sequence: Some(i as u32),
-        })
-        .collect()
+/// Creates an OutputGroup from a transaction input and its value
+/// Value would normally be queried from the mainchain for the UTXO being spent
```

Comment:
```suggestion
/// Value would normally be queried from the mainchain for the UTXO being spent
```
/// In practice, the details about the UTXO, used as input, is obtained from the UTXO set maintained by a node.

Reply:
Done